service/history/workflow/context.go

```diff
@@ -140,10 +140,24 @@ type (
 			updateWorkflowTransactionPolicy TransactionPolicy,
 			newWorkflowTransactionPolicy *TransactionPolicy,
 		) error
+		// SetWorkflowExecution is an alias to SubmitClosedWorkflowMutation with TransactionPolicyPassive.
 		SetWorkflowExecution(
 			ctx context.Context,
 			shardContext shard.Context,
 		) error
+		// SubmitClosedWorkflowMutation closes the current mutable state transaction with the given
+		// transactionPolicy and updates the workflow execution record in the DB. Does not check the "current"
+		// run status for the execution.
+		// Closes the transaction as snapshot, which errors out if there are any buffered events that need
+		// flushing and generally does not expect new history events to be generated (expected for closed
+		// workflows).
+		// NOTE: in the future, we'd like to have the ability to close the transaction as mutation to avoid the
+		// overhead of overwriting the entire DB record.
+		SubmitClosedWorkflowMutation(
```

Comment:
Yes, sure.
We'll want to refactor this later all the way to persistence to support closing as mutation though to avoid writing all of MS. We can change the name of this API when persistence supports this.